src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs

```diff
@@ -646,6 +656,18 @@ private void HideProgress()
                 (uint)100);
         }
 
+        private async Task HideProgressAsync()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
This breaks vs threading rule #1. 

async methods transition to the ui thread.

Reply:
Ok. Addressed it.